plugins/modules/lxd_storage_volume_info.py

```diff
@@ -0,0 +1,402 @@
+    }
+  ]
+  contains:
+    name:
+      description: The name of the storage volume.
+      type: str
+      returned: success
+    type:
+      description: The type of the storage volume.
+      type: str
+      returned: success
+    used_by:
+      description: List of resources using this storage volume.
+      type: list
+      elements: str
+      returned: success
+    config:
+      description: Configuration of the storage volume.
+      type: dict
+      returned: success
+    description:
+      description: Description of the storage volume.
+      type: str
+      returned: success
+    content_type:
+      description: Content type of the volume (filesystem or block).
+      type: str
+      returned: success
+    location:
+      description: Cluster member location for the volume.
+      type: str
+      returned: success
+logs:
+  description: The logs of requests and responses.
+  returned: when ansible-playbook is invoked with -vvvv.
+  type: list
+  elements: dict
+  sample: [
+    {
+      "type": "sent request",
+      "request": {
+        "method": "GET",
+        "url": "/1.0/storage-pools/default/volumes",
+        "json": null,
+        "timeout": null
+      },
+      "response": {
+        "json": {"type": "sync", "status": "Success"}
+      }
+    }
+  ]
+"""
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible_collections.community.general.plugins.module_utils.lxd import (
+    LXDClient,
+    LXDClientException,
+    default_cert_file,
+    default_key_file,
+)
+
+# ANSIBLE_LXD_DEFAULT_URL is a default value of the lxd endpoint
+ANSIBLE_LXD_DEFAULT_URL = "unix:/var/lib/lxd/unix.socket"
+
+
+class LXDStorageVolumeInfo(object):
+    def __init__(self, module):
+        """Gather information about LXD storage volumes.
+
+        :param module: Processed Ansible Module.
+        :type module: ``object``
+        """
+        self.module = module
+        self.pool = self.module.params["pool"]
+        self.name = self.module.params["name"]
+        self.volume_type = self.module.params["type"]
+        self.project = self.module.params["project"]
+
+        self.key_file = self.module.params["client_key"]
+        if self.key_file is None:
+            self.key_file = default_key_file()
+        self.cert_file = self.module.params["client_cert"]
+        if self.cert_file is None:
+            self.cert_file = default_cert_file()
+        self.debug = self.module._verbosity >= 4
+
+        try:
+            if self.module.params["url"] != ANSIBLE_LXD_DEFAULT_URL:
+                self.url = self.module.params["url"]
+            elif os.path.exists(self.module.params["snap_url"].replace("unix:", "")):
+                self.url = self.module.params["snap_url"]
+            else:
+                self.url = self.module.params["url"]
+        except Exception as e:
+            self.module.fail_json(msg=str(e))
```

Comment:
Same comments made in one module are likely to apply to the other. Please review the changes requested so far and make sure both modules are consistent.

(This marked block here is where we removed the redundant try/except in the other module - my apologies, I assumed it was implicit that the change applied to both modules).